### ISSUE TITLE
Refactor Flags

### ISSUE THREAD

**Issue by duncanspumpkin** [OpenRCT2/OpenRCT2#5173]
Go through the code base and try use the appropriate flags enums.

My technique is to search for `flags & 0x` and see if any of the hits have a flags enum. See https://github.com/OpenRCT2/OpenRCT2/pull/5172 for an example of this. 

Note some flags are in the form of `(1 << X)`. Try use the appropriate enum for them as well.

Occasionally I've encountered a flags without an enum. Create an appropriate enum for these.

**Comment by platipo** [comment]
Does `#define DEFINE (1 << n)` count too?

**Comment by Gymnasiast** [comment]
@platipo If it's really a _flag_, it should indeed be in an enum.

**Comment by platipo** [comment]
@Gymnasiast I will provvide and example, tell me if it shold be an enum or not. 
[Example 1](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/management/marketing.h#L41)

Also must all the following directories be refractored? 
 -	openrct2-android 
-	openrct2-cli 	
-	openrct2-dll 	
-	openrct2-ui 	
-	openrct2-win 	
-	openrct2

**Comment by platipo** [comment]
The enum definition should be in the file itself or in the header?
And how are names picked?

**Comment by Gymnasiast** [comment]
>  will provide some examples, tell me if they should be an enum or not.

Example 1: yes. Example 2: yes, including the three flags following it.

> Also must all the following directories be refactored?

Yes.

> The enum definition should be in the file itself or in the header?

In the header.

> And how are names picked?

![Magic](http://s2.quickmeme.com/img/72/7221f725e518793a71eaec4da3d0d5b60fa961b374f3b33f8a83867da2dc331e.jpg)

Seriously though, it's hard to answer. I would suggest looking at other flag enums.

**Comment by platipo** [comment]
I'm fairly new to the codebase, so I think it would be easier mark each `flag`, `0x` and `0b` to refractor leaving the naming to someone more experienced. I'll add `\* TODO: refractor flag *\` next to ambigous nameless definitions; in case of `#define` I'll create the `enum`.
```bash
FILES=$(find . -type f)
for F in $FILES; do
        FOUND=$(nl -ba $F | egrep "TODO: refractor flag")
        if [ -n "$FOUND" ]; then
                echo "==="$F"===\n""$FOUND"
        fi
done
```
What do you think @Gymnasiast?

**Comment by Gymnasiast** [comment]
There are lots of places where hexadecimal or binary are used as non-flags. I doubt that plan would really save us time, if I'm brutally honest.

Moving existing flag defines to enums would help, but you would need to look for the appropriate header file. If you're unsure: edit one file, then open a 'WIP' pull request and we can look at it in more detail.

Finally, some small things: 
1. It's refactoring, not 'refractoring'.
2/ Comments use forward slashes, not backslashes. Backslashes are only used for escaping.

**Comment by platipo** [comment]
I agree with your opinion but still naming is a problem for me, it's a lot harder choose a consistent name without knowing the codebase, don't you think?

**Comment by Gymnasiast** [comment]
If you really can't think of a name, create an enum with entries like
MAZE_ENTRY_FLAG_15. We have done that ourselves in the past, and renaming is easy for us.

**Comment by platipo** [comment]
Thanks! One last question (I hope) in the last example is it still required that the enum is created in a header or can I create it in the same file?

**Comment by platipo** [comment]
@Gymnasiast [Am I on the right track?](https://github.com/platipo/OpenRCT2/commits/flags-refactor)

**Comment by platipo** [comment]
@Gymnasiast I suppose I'm not seeing the difference between a a flag and a non-flag. Can you explain the difference once more?

**Comment by Gymnasiast** [comment]
Flags are powers of two: 0, 1, 2, 4, 8, etc. Usually they're written as bitshifted values, like (1 << 3). Keep in mind that (1 << n) = 2^n. Example: (1 << 3) = 2^3 = 8.

**Comment by platipo** [comment]
@Gymnasiast I see lots of 1,2,4,8 linked to slopes ([expecially](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/world/map_helpers.c)) but find no explicit flag related

Thanks for the explanation, can you tell me if the last two commits should be dropped?
- [1](https://github.com/platipo/OpenRCT2/commit/ceaec7d68f3c34894e2d2160225a848003fb65c1)
- [2](https://github.com/platipo/OpenRCT2/commit/a28b028b8fc21b4591f85bef0219f939553ae761)

**Comment by Gymnasiast** [comment]
No.

**Comment by duncanspumpkin** [comment]
Just in case anyone is still wondering. There are still quite a few flags to refactor. We are at least getting to the more obscure ones at least.

**Comment by Gymnasiast** [comment]
Some constants to address:

- [ ] `scrolling_mode = 0xFF` -> `scrolling_mode = SCROLLING_MODE_NONE`

**Comment by InsigMath** [comment]
I saw a few potential flags that could be refactored, would it be possible to be assigned it
